plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/state/FollowingViewportStateImpl.kt

```diff
@@ -0,0 +1,238 @@
+package com.mapbox.maps.plugin.viewport.state
+
+import android.animation.Animator
+import android.animation.AnimatorSet
+import android.animation.ValueAnimator
+import androidx.core.animation.doOnEnd
+import com.mapbox.common.Logger
+import com.mapbox.geojson.Point
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.plugin.animation.Cancelable
+import com.mapbox.maps.plugin.animation.camera
+import com.mapbox.maps.plugin.delegates.MapDelegateProvider
+import com.mapbox.maps.plugin.locationcomponent.LocationComponentPlugin
+import com.mapbox.maps.plugin.locationcomponent.OnIndicatorBearingChangedListener
+import com.mapbox.maps.plugin.locationcomponent.OnIndicatorPositionChangedListener
+import com.mapbox.maps.plugin.locationcomponent.location
+import com.mapbox.maps.plugin.viewport.data.FollowingViewportStateBearing
+import com.mapbox.maps.plugin.viewport.data.FollowingViewportStateOptions
+import com.mapbox.maps.plugin.viewport.transition.MapboxViewportTransitionFactory
+import java.util.concurrent.*
+
+/**
+ * The actual implementation of [FollowingViewportState] that follows user's location.
+ *
+ * Note: [LocationComponentPlugin] should be enabled to use this viewport state.
+ */
+class FollowingViewportStateImpl internal constructor(
```

Comment:
same as above - shouldn't we mark our impl as `internal`?

Reply:
changed to internal class.